tests/ClientTest.php

```diff
@@ -534,43 +357,268 @@ public function testAddBreadcrumbStoresBreadcrumbReturnedByBeforeBreadcrumbCallb
+        $malformedString = "\xC2\xA2\xC2"; // ill-formed 2-byte character U+00A2 (CENT SIGN)
+
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function (Event $event) {
+                $this->assertNotNull($event);
+
+                $expectedValue = [
+                    [
+                        'type' => \Exception::class,
+                        'value' => "\xC2\xA2\x3F",
+                    ],
+                ];
+
+                $this->assertArraySubset($expectedValue, $event->getException());
+
+                return true;
+            }));
+
+        $client = new Client(new Options(), $transport, []);
+        $client->captureException(new \Exception($malformedString));
+    }
+
+    public function testConvertExceptionThrownInLatin1File()
+    {
+        $options = new Options([
+            'auto_log_stacks' => true,
+            'mb_detect_order' => ['ISO-8859-1', 'ASCII', 'UTF-8'],
+        ]);
+
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function (Event $event) {
+                $this->assertNotNull($event);
+
+                $result = $event->getException();
+                $expectedValue = [
+                    [
+                        'type' => \Exception::class,
+                        'value' => 'foo',
+                    ],
+                ];
+
+                $this->assertArraySubset($expectedValue, $result);
+
+                $latin1StringFound = false;
+
+                /** @var \Sentry\Frame $frame */
+                foreach ($result[0]['stacktrace']->getFrames() as $frame) {
+                    if (null !== $frame->getPreContext() && \in_array('// äöü', $frame->getPreContext(), true)) {
+                        $latin1StringFound = true;
+
+                        break;
+                    }
+                }
+
+                $this->assertTrue($latin1StringFound);
+
+                return true;
+            }));
+
+        $client = new Client($options, $transport, []);
+        $client->captureException(require_once __DIR__ . '/Fixtures/code/Latin1File.php');
+    }
+
+    public function testConvertExceptionWithAutoLogStacksDisabled()
+    {
+        $options = new Options(['auto_log_stacks' => false]);
+
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function (Event $event) {
+                $this->assertNotNull($event);
```

Comment:
This assert is unnecessary because the argument is typehinted to not accept `null` values